apps/vaults/contexts/useActionFlow.tsx

```diff
@@ -0,0 +1,226 @@
+	onUpdateSelectedOptionFrom: (option: TDropdownOption) => void;
+	onUpdateSelectedOptionTo: (option: TDropdownOption) => void;
+	onSwitchSelectedOptions: () => void;
+	isDepositing: boolean;
+	maxDepositPossible: TNormalizedBN;
+	currentSolver: Solvers;
+}
+const	DefaultActionFlowContext: TActionFlowContext = {
+	currentVault: {} as TYearnVault, // eslint-disable-line @typescript-eslint/consistent-type-assertions
+	possibleOptionsFrom: [],
+	possibleOptionsTo: [],
+	selectedOptionFrom: undefined,
+	selectedOptionTo: undefined,
+	amount: DefaultTNormalizedBN,
+	onChangeAmount: (): void => undefined,
+	onUpdateSelectedOptionFrom: (): void => undefined,
+	onUpdateSelectedOptionTo: (): void => undefined,
+	onSwitchSelectedOptions: (): void => undefined,
+	isDepositing: true,
+	maxDepositPossible: DefaultTNormalizedBN,
+	currentSolver: Solvers.VANILLA
+};
+
+const ActionFlowContext = createContext<TActionFlowContext>(DefaultActionFlowContext);
+function ActionFlowContextApp({children, currentVault}: {children: ReactNode, currentVault: TYearnVault}): React.ReactElement {
+	const {balances} = useWallet();
+	const {safeChainID} = useChainID();
+	const {networks} = useSettings();
+	const {balances: zapBalances, tokensList} = useWalletForZap();
+
+	const [possibleOptionsFrom, set_possibleOptionsFrom] = useState<TDropdownOption[]>([]);
+	const [possibleZapOptionsFrom, set_possibleZapOptionsFrom] = useState<TDropdownOption[]>([]);
+	const [possibleOptionsTo, set_possibleOptionsTo] = useState<TDropdownOption[]>([]);
+	const [selectedOptionFrom, set_selectedOptionFrom] = useState<TDropdownOption | undefined>();
+	const [selectedOptionTo, set_selectedOptionTo] = useState<TDropdownOption | undefined>();
+	const [amount, set_amount] = useState<TNormalizedBN>(DefaultTNormalizedBN);
+
+	const isDepositing = useMemo((): boolean => (
+		!selectedOptionTo?.value ? true : toAddress(selectedOptionTo.value) === toAddress(currentVault.address)
+	), [selectedOptionTo, currentVault]);
+
+	const isPartnerAddressValid = useMemo((): boolean => (
+		!isZeroAddress(toAddress(networks?.[safeChainID]?.partnerContractAddress))
+	), [networks, safeChainID]);
+
+	const isUsingPartnerContract = useMemo((): boolean => (
+		(process?.env?.SHOULD_USE_PARTNER_CONTRACT === undefined ? true : Boolean(process?.env?.SHOULD_USE_PARTNER_CONTRACT)) && isPartnerAddressValid
+	), [isPartnerAddressValid]);
+
+	const maxDepositPossible = useMemo((): TNormalizedBN => {
+		const	vaultDepositLimit = formatBN(currentVault.details.depositLimit) || ethers.constants.Zero;
+		const	userBalance = balances?.[toAddress(selectedOptionFrom?.value)]?.raw || ethers.constants.Zero;
+		if (userBalance.gt(vaultDepositLimit)) {
+			return ({
+				raw: vaultDepositLimit,
+				normalized: formatToNormalizedValue(vaultDepositLimit, currentVault.token.decimals)
+			});
+		} else {
+			return ({
+				raw: userBalance,
+				normalized: balances?.[toAddress(selectedOptionFrom?.value)]?.normalized || 0
+			});
+		}
```

Comment:
I think the less `else`s the better :)
```suggestion
				}

		return ({
			raw: userBalance,
			normalized: balances?.[toAddress(selectedOptionFrom?.value)]?.normalized || 0
		});
```

Reply:
Excellent! 👍